rxandroidble/src/test/groovy/com/polidea/rxandroidble/RxBleClientTest.groovy

```diff
@@ -79,6 +79,21 @@ class RxBleClientTest extends Specification {
         1 * bleAdapterWrapperSpy.startLeScan(_) >> true
     }
 
+    def "should start BLE scan if bluetooth is enabled before subscribing to the scan observable"() {
+        given:
+        TestSubscriber testSubscriber = new TestSubscriber<>()
+        bleAdapterWrapperSpy.hasBluetoothAdapter() >> true
+        bleAdapterWrapperSpy.isBluetoothEnabled() >> false
+
+        when:
```

Comment:
@dariuszseweryn Will this approach be fine? 

      
        given:
        TestSubscriber testSubscriber = new TestSubscriber<>()
        bleAdapterWrapperSpy.hasBluetoothAdapter() >> true
        bleAdapterWrapperSpy.isBluetoothEnabled() >> true

        when:
        def scanObservable = objectUnderTest.scanBleDevices(null)

        then:
        0 * bleAdapterWrapperSpy.isBluetoothEnabled()

        when:
        scanObservable.subscribe(testSubscriber)

        then:
        1 * bleAdapterWrapperSpy.startLeScan(_) >> true